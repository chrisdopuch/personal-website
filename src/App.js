import React, { Component } from 'react';
import me from './me.jpg';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={me} className="App-face" alt="my-face"/>
          <p>
            I wrote enough React to make my head spin!
          <br/>
            More content to come soon...
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

export default App;
