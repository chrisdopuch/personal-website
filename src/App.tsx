// import AppBar from '@material-ui/core/AppBar';
import CssBaseline from "@material-ui/core/CssBaseline";
// import IconButton from '@material-ui/core/IconButton';
import { createStyles, Theme, withStyles, WithStyles } from "@material-ui/core/styles";
// import Toolbar from '@material-ui/core/Toolbar';
// import Typography from '@material-ui/core/Typography';
// import MenuIcon from '@material-ui/icons/Menu';
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import ResponsiveDrawer from "./components/ResponsiveDrawer";
// import me from './images/me.jpg';

const stylesDeclarations = (theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      textAlign: "center",
    },
  });

interface IAppProps extends WithStyles<typeof stylesDeclarations> {
  title: string;
  theme: Theme;
}

class App extends React.Component<IAppProps> {
  public render() {
    const { classes, title } = this.props;
    return (
      <React.Fragment>
        <CssBaseline />
        <Router>
          <div className={classes.root}>
            <ResponsiveDrawer title={title} />
          </div>
        </Router>
      </React.Fragment>
    );
  }
}

export default withStyles(stylesDeclarations, { withTheme: true })(App);
