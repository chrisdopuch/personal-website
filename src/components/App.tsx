import CssBaseline from '@material-ui/core/CssBaseline';
import { createStyles, MuiThemeProvider, Theme, WithStyles, withStyles } from '@material-ui/core/styles';
import CameraIcon from '@material-ui/icons/Camera';
import HelpIcon from '@material-ui/icons/Help';
import HomeIcon from '@material-ui/icons/Home';
import React from 'react';
import { useStore } from 'react-hookstore';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { IAppStore } from '../store';
import { appTheme, darkTheme } from '../themes';
import NavDrawer from './NavDrawer';
import About from './pages/About';
import Gallery from './pages/Gallery';
import Home from './pages/Home';
import TitleBar from './TitleBar';

const navBarItems = [
  { to: '/', label: 'Home', Icon: HomeIcon, isRouteExact: true, page: Home },
  { to: '/about', label: 'About me', Icon: HelpIcon, isRouteExact: false, page: About },
  { to: '/gallery', label: 'Gallery', Icon: CameraIcon, isRouteExact: false, page: Gallery },
];

const stylesDeclarations = (theme: Theme) => {
  return createStyles({
    content: {
      flexGrow: 1,
      padding: theme.spacing.unit * 3,
    },
    root: {
      display: 'flex',
      flexGrow: 1,
      textAlign: 'center',
    },
    toolbar: theme.mixins.toolbar,
  });
};

interface IAppProps extends WithStyles<typeof stylesDeclarations, true> {
  title: string;
}

export const App: React.SFC<IAppProps> = (props) => {
  const { classes, title } = props;
  const [appStore, _setStore]: [IAppStore, (s: IAppStore) => void] = useStore('appStore');

  return (
    <MuiThemeProvider theme={appStore.isDarkMode ? darkTheme : appTheme}>
      <CssBaseline />
      <Router>
        <div className={classes.root}>
          <TitleBar title={title} />
          <NavDrawer items={navBarItems} />
          <main className={classes.content}>
            <div className={classes.toolbar} />
            {navBarItems.map(({ to, page, isRouteExact }) => {
              return <Route path={to} component={page} exact={isRouteExact} key={to} />;
            })}
          </main>
        </div>
      </Router>
    </MuiThemeProvider>
  );
};

export default withStyles(stylesDeclarations, { withTheme: true })(App);
