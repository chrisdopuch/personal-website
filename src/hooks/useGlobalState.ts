import { useContext } from 'react';
import { stateCtx } from '../components/contexts/state';
import { State } from '../store';

const useGlobalState = <K extends keyof State>(key: K) => {
  const state = useContext(stateCtx);
  return state[key];
};

export default useGlobalState;
