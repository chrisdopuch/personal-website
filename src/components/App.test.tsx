import { Theme } from '@material-ui/core';
import { shallow } from 'enzyme';
import React from 'react';
import useGlobalState from '../hooks/useGlobalState';
import { App } from './App';

jest.mock('../hooks/useGlobalState', () => {
  return jest.fn(() => false);
});

jest.mock('../themes', () => ({ darkTheme: {}, appTheme: {} }));

function getDefaultProps() {
  return {
    classes: {
      content: 'content',
      root: 'root',
      toolbar: 'toolbar',
    },
    theme: {} as Theme,
    title: 'test',
  };
}

describe('App', () => {
  it('renders with default props', () => {
    const wrapper = shallow(<App {...getDefaultProps()} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders when `isDarkMode` is true', () => {
    (useGlobalState as jest.Mock).mockImplementationOnce(() => true);
    const wrapper = shallow(<App {...getDefaultProps()} />);

    expect(wrapper).toMatchSnapshot();
  });
});
