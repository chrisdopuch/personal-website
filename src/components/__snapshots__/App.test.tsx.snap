// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders Projects 1`] = `
<WithStyles(Typography)
  component="h3"
  variant="h2"
>
  Projects
</WithStyles(Typography)>
`;

exports[`App renders default props 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <MuiThemeProviderOld
    theme={
      Object {
        "breakpoints": Object {
          "up": [MockFunction],
        },
        "spacing": Object {
          "unit": 1,
        },
        "typography": Object {
          "useNextVariants": true,
        },
        "zIndex": Object {
          "drawer": 1,
        },
      }
    }
  >
    <BrowserRouter>
      <div
        className="root"
      >
        <WithStyles(TitleBar)
          title="test"
        />
        <WithStyles(NavDrawer)
          items={
            Array [
              Object {
                "Icon": [Function],
                "isRouteExact": true,
                "label": "Home",
                "page": [Function],
                "to": "/",
              },
              Object {
                "Icon": [Function],
                "isRouteExact": false,
                "label": "About me",
                "page": [Function],
                "to": "/about",
              },
              Object {
                "Icon": [Function],
                "isRouteExact": false,
                "label": "Gallery",
                "page": [Function],
                "to": "/gallery",
              },
              Object {
                "Icon": [Function],
                "isRouteExact": false,
                "label": "Projects",
                "page": [Function],
                "to": "/projects",
              },
            ]
          }
        />
        <main
          className="content"
        >
          <div
            className="toolbar"
          />
          <Route
            component={[Function]}
            exact={true}
            key="/"
            path="/"
          />
          <Route
            component={[Function]}
            exact={false}
            key="/about"
            path="/about"
          />
          <Route
            component={[Function]}
            exact={false}
            key="/gallery"
            path="/gallery"
          />
          <Route
            component={[Function]}
            exact={false}
            key="/projects"
            path="/projects"
          />
        </main>
      </div>
    </BrowserRouter>
  </MuiThemeProviderOld>
</Fragment>
`;
